src/tardev-snapshotter/docs/ROADMAP.md

```diff
@@ -0,0 +1,148 @@
+# Code Integrity for OCI Containers Roadmap
+
+The purpose of this issue is to track the roadmap for the Code Integrity for OCI
+Containers project.
+
+The goal is to ensure that when a container is running, its integrity is
+enforced by kernel in the runtime
+([dm-verity](https://www.kernel.org/doc/html/latest/admin-guide/device-mapper/verity.html))
+and further, only binaries that come from the original container image layers
+are allowed to run
+([IPE](https://docs.kernel.org/next/admin-guide/LSM/ipe.html)). Finally, we
+would like to achieve these goals seamlessly, such that not everybody is
+required to update the build and runtime environments, by leveraging integration
+with OCI registries.
+
+The rest of the document outlines the milestones we envision to achieve the
+stated goals.
+
+## Milestones
+
+### Milestone 1: Initial Integration with containerd/erofs-snapshotter
+
+The purpose of this milestone is to enable the basic building blocks required
+for the Code Integrity for OCI Containers project. This includes support for tar
+index, dm-verity, and dm-verity signatures in the erofs-snapshotter.
+
+As part of the initial milestone, we would like to extend the
+erofs-snapshotter with the following features:
+
+1. Add a second method for image layer tar to erofs conversion in the erofs
+   snapshotter by leveraging the tar index mode. This is a prerequisite for the
+   next milestone, so that the index, which is much smaller than the full erofs
+   blob, can be generated ahead of time and pulled from the registry (instead of
+   being generated on the fly on the container host). It will also serve as a
+   consistent fallback for the next milestone, in case the tar index is not
+   available in the registry.
+
+2. Finalize the dm-verity support in the erofs
+   snapshotter, so that the converted image layers can be backed by a dm-verity
+   block device. This will allow the kernel to check the integrity of any file that
+   is part of image layers in the runtime. As part of the initial milestone, the
+   dm-verity Merkle tree will be generated on the fly on the container host.
+
+3. Extend the dm-verity support with logic to pull per-layer dm-verity root hash
+   signatures. This will establish provenance for each layer, recognized by the
+   kernel. This is a prerequisite for leveraging IPE at scale for all the
+   container image layers. We envision that the
+   signatures would be pulled from the registry and tied to image layers by
+   leveraging the OCI referrers feature. Tying the signatures to the layers
+   directly, as opposed to the image manifest, will allow the signatures to be
+   reused across different image manifests.
+
+   ```mermaid
+   erDiagram
+     direction LR
+     IMAGE-MANIFEST }|--|{ IMAGE-LAYER-TAR: layers
```

Comment:
I understand the concept here, but what does `}|--|{`, `||--||`, `||..|{` show? Have not seen many mermaid diagrams